lib/jekyll/commands/serve.rb

```diff
@@ -33,6 +33,8 @@ def init_with_program(prog)
               opts["serving"] = true
               opts["watch"  ] = true unless opts.key?("watch")
               config = opts["config"]
+              default_url = "http://localhost:4000"
```

Comment:
One can set the `host` and `port` options in the command line or via your config file. If I set my `--port` to `3999`, I should get my default url as `http://localhost:3999`. Same with `--host`.